github/git_refs.go

```diff
@@ -159,10 +161,7 @@ func (s *GitService) CreateRef(ctx context.Context, owner, repo string, ref *Ref
 // GitHub API docs: https://docs.github.com/rest/git/refs#update-a-reference
 //
 //meta:operation PATCH /repos/{owner}/{repo}/git/refs/{ref}
-func (s *GitService) UpdateRef(ctx context.Context, owner, repo string, ref *Reference, force bool) (*Reference, *Response, error) {
-	if ref == nil {
-		return nil, nil, errors.New("reference must be provided")
-	}
+func (s *GitService) UpdateRef(ctx context.Context, owner, repo string, ref Reference, force bool) (*Reference, *Response, error) {
```

Comment:
The `Reference` has 4 fields, but this method used only two: `ref (string)` and `sha (string)`. I propose to create a separate struct similar to what we already did for `CreateRef`:

```suggestion
func (s *GitService) UpdateRef(ctx context.Context, owner, repo, ref string, updateRef UpdateRef) (*Reference, *Response, error) {
```

And this struct should be used instead of `updateRefRequest`:
```go
type UpdateRef struct {
	SHA   string `json:"sha"`
	Force *bool   `json:"force,omitempty"`
}
```